src/NuGet.Core/NuGet.Common/RuntimeEnvironmentHelper.cs

```diff
@@ -37,6 +39,34 @@ public static class RuntimeEnvironmentHelper
         [DllImport("libc")]
         static extern int uname(IntPtr buf);
 
+        static string GetUnixName()
+        {
+            var buf = IntPtr.Zero;
+
+            try
+            {
+                buf = Marshal.AllocHGlobal(8192);
+
+                // This is a hacktastic way of getting sysname from uname ()
+                if (uname(buf) == 0)
```

Comment:
Why not use [OperatingSystem.IsOSPlatform](https://docs.microsoft.com/en-us/dotnet/api/system.operatingsystem.isosplatform?view=net-6.0#system-operatingsystem-isosplatform(system-string)) and [OSPlatform.FreeBSD](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.osplatform.freebsd?view=net-6.0#system-runtime-interopservices-osplatform-freebsd)?

OSPlatformFreeBSD only exists in .NET Core 3.1 and higher, whereas this project is also compiled for .NET Standard and .NET Framework, so there will need to be an `#if`. But I would prefer to use BCL methods rather than P/Invoke, especially considering it involves `Marshal.AllocHGlobal`.

Reply:
> But I would prefer to use BCL methods rather than P/Invoke, especially considering it involves `Marshal.AllocHGlobal`.

I did a little research to figure out [how the .NET runtime gets the Unix name](https://github.com/dotnet/runtime/blob/b1edcf125122488e1be32ffce54d909e99825d2d/src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetUnixName.cs) to [check on which OS platform the current application is running](https://github.com/dotnet/runtime/blob/b1edcf125122488e1be32ffce54d909e99825d2d/src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs#L93-L115) and adopted it.